packages/runtime/lifecycle/src/api.ts

```diff
@@ -36,12 +37,34 @@ export type RuntimeLifecycle = {
  */
 export type ContractsDI = WalletDI & RestDI;
 
-export type CreateContractRequest = {
+/**
+ * Request parameters used by {@link api.ContractsAPI#createContract | createContract}.
+ * If the contract is "small", you can create it directly with a {@link CreateContractRequestFromContract| core contract},
+ * if the contract is "large" you can use a {@link CreateContractRequestFromBundle | contract bundle} instead.
+ * Both options share the same {@link CreateContractRequestBase | request parameters}.
+
+ */
+export type CreateContractRequest =
+  | CreateContractRequestFromContract
+  | CreateContractRequestFromBundle;
+
+export interface CreateContractRequestFromContract
+  extends CreateContractRequestBase {
```

Comment:
Not sure, this pattern is better than a `ContractOrSourceId` parameter, I'd prefer also `Composition` over `Inheritance` ... and the doc is probably less readable that way...  but ok, we can always change it later when we'll add more functionalities...

Reply:
take a look at the generated docs, starting from the function. I think it ended up being readable.

IMO, Composition vs Inheritance does not apply to this. I'm not doing class inheritance